python/cugraph/cugraph/structure/graph_classes.py

```diff
@@ -683,6 +683,27 @@ def add_nodes_from(self, nodes):
         """
         self._Impl._nodes["all_nodes"] = cudf.Series(nodes)
 
+    def density(self):
+        """
+        Return the density of the graph.
+        Density is the measure of how many edges are in the graph versus
+        the max number of edges that could be present.
+        Note:  This function does not work on multigraph since there
+        is no theoretical max number of edges
+        """
+        if self.is_multigraph():
+            raise TypeError("G cannot be a Multigraph")
```

Comment:
This should be removed and the `MultiGraph` class should instead add a `density` method which raises this exception. That will also allow you to cleanup the docstring and mention the MultiGraph specifics there.

Also, the message shouldn't mention `G` since it's not used here and could be confusing (eg. a user has a variable named `G` in their code and they get this message).

Reply:
moved to MultiGraph section